src/Network/GRPC/MQTT/Message/Packet.hs

```diff
@@ -158,55 +164,110 @@ makePacketReader queue = do
 
 -- Packet - Packet Senders -----------------------------------------------------
 
+newtype PacketSizeError = PacketSizeError
+  {getPacketSizeError :: Integer}
+  deriving (Data, Eq, Generic, Ord, Show, Typeable)
+
+-- | @since 1.0.0
+instance Exception PacketSizeError where
+  displayException (PacketSizeError n) =
+    Text.printf
+      "%s: maximum packet payload size must be between %d and %d bytes."
+      (show (PacketSizeError n) :: String)
+      minPacketSize
+      maxPacketSize
+  {-# INLINE displayException #-}
+
+-- | The predicate 'isValidPacketSize' tests if the 'Word32' can be used as the
+-- maximum packet payload size in 'makePacketSender'. This is any 'Word32' value
+-- that is:
+--
+-- * Greater than or equal to 'minPacketSize'.
+--
+-- * Strictly less than 'maxPacketSize'.
+--
+-- @since 1.0.0
+isValidPacketSize :: Word32 -> Bool
+isValidPacketSize x = minPacketSize <= x && x < maxPacketSize
+
+-- | The minimum packet payload size, 16 bytes.
+--
+-- @since 1.0.0
+minPacketSize :: Word32
+minPacketSize = 16
+
+-- | The maximum packet payload size, 256mB.
+--
+-- @since 1.0.0
+maxPacketSize :: Word32
+maxPacketSize = 256 * 2 ^ (20 :: Int)
```

Comment:
Where does this limit come from?

Reply:
256mB is the maximum size for a MQTT publish.